### ISSUE TITLE
scale_{colour,fill}_manual(limits = ...) doesn't limit the scale

### ISSUE THREAD

**Issue by yutannihilation** [tidyverse/ggplot2#3262]
(I found this during looking at the examples on [`scale_manual`](https://ggplot2.tidyverse.org/reference/scale_manual.html#examples).)

It seems `scale_{colour,fill}_manual(limits = ...)` doesn't limit the scale actually (but limit the legend).

``` r
library(ggplot2)
library(patchwork)

p <- ggplot(mtcars, aes(factor(cyl))) +
  geom_bar(aes(fill = factor(cyl)))

# this drops 6
p1 <- p + scale_fill_discrete(limits = c("4", "8"))

# this doesn't
cols <- c("8" = "red", "4" = "blue", "6" = "darkgreen", "10" = "orange")
p2 <- p + scale_fill_manual(values = cols, limits = c("4", "8"))

p1 / p2
```

![](https://i.imgur.com/C1x2RgM.png)

<sup>Created on 2019-04-24 by the [reprex package](https://reprex.tidyverse.org) (v0.2.1)</sup>

**Comment by thomasp85** [comment]
Do you think we can fix this for 3.2.0?

**Comment by yutannihilation** [comment]
I forgot to reply here. After a quick look, I think there are two possible fixes:

1. Add `limits` argument to `manual_scale()` and subset `values` before creating a palette from the values.
2. Raise an error when `limits` is provided.

First one seems straightforward, but I enumerate the second because `manual_scale()` can accept `pal` while it's difficult to limit the `pal` and it might be confusing.

If we go with the first, probably it will be something like this:

``` r
manual_scale <- function(aesthetic, values = NULL, ..., limits = NULL) {
  ...
  if (!is.null(limits)) {
    values <- values[values %in% limits]
  }

  pal <- function(n) {
  ...
```
